### ISSUE TITLE
Add a sample of Projection with MongoDb Provider

### ISSUE THREAD

**Issue by vinicius-actcon** [ChilliCream/hotchocolate-examples#39]
Hello everyone!

Thanks for became this amazing solution avaliable for the  .NET ecosystem.

I"m reading the documentation of "[how use HotChocolate with MongoDb at projection level](https://chillicream.com/docs/hotchocolate/integrations/mongodb#projections)" but I did found any reference of how to use the projection itself.

For exemple, i see that after IMongoCollection<T> we have access to .AsExecutable() extension method and afterwards this execution whe hav access to *With* options (WithSorting|WithFiltering|WithOptions|WithProjection) and WithProjection method expect an argument of type MongoDbProjectionDefinition, but how can i instantiate this definition?

There's no one reference in nowhere how to create(instantiate) that definition to be used by projection. Could someone provide a sample of how to do that?

Thanks!

**Comment by vinicius-actcon** [comment]
yeah, I'm aware about the performance issues with use of projections, but I really need a projection due to not expose all fields for a given entity.

How could create an "projection" of a entity type with an model type?

**Comment by PascalSenn** [comment]
@vinicius-actcon  hmm not sure what you mean.
